### PR TITLE
RavenDB-16409 - Client.TimeSeries.BulkInsert.CanAppendALotOfValuesAnd…

### PR DESCRIPTION
**Pull request by shiranshalom**
…TimeSeriesInParallel - multiple errors

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-16409

### Additional description

Failed to reproduce the issue on Ubuntu 20.04.4 VM, .NET 6 framework. 
Adding additional debug data. 

Seems that this test fails only on `Testsv52_Winx64_Debug` version on both machines - Eagle 2 & 4. 
In addition, there are no common properties that I found on previous fails (for example, failing at the same time each run). 

My investigation so far lead me to think that an exception was thrown in `For.Parallel` section but we missed catching it due to the '_**async**_' lambda when the test return '_**void**_' type.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: test/SlowTests/Client/TimeSeries/BulkInsert/BulkInsert.cs [hunk 1]
```diff
@@ -1124,30 +1124,45 @@ public void CanAppendALotOfValuesAndTimeSeriesInParallel(int numberOfValues)
                     session.SaveChanges();
                 }
                 var count = new CountdownEvent(numberOfTimeSeries);
+
+                var debugData = new string[numberOfTimeSeries];
+
                 Parallel.For(0, numberOfTimeSeries, async (i) =>
```
Comment (ayende) [inline test/SlowTests/Client/TimeSeries/BulkInsert/BulkInsert.cs]: Pretty sure that this is wrong, the `async` on the `Paralle.For`, who waits for this to complete?

Comment (shiranshalom) [reply test/SlowTests/Client/TimeSeries/BulkInsert/BulkInsert.cs]: Not sure I follow. We are waiting on `count.Wait(TimeSpan.FromMinutes(5))`

Comment (ayende) [reply test/SlowTests/Client/TimeSeries/BulkInsert/BulkInsert.cs]: It bugs me that we are doing it in this manner. It _works_, but it isn't a good way to handle this.
Bad pattern for multi thread ing / syncronizations

Comment (ppekrol) [reply test/SlowTests/Client/TimeSeries/BulkInsert/BulkInsert.cs]: Probably we can switch to `Parallel.ForEachAsync(Enumerable.Range(0, numberOfTimeSeries), (i, token) => ...)`

Comment (shiranshalom) [reply test/SlowTests/Client/TimeSeries/BulkInsert/BulkInsert.cs]: done

**Pull request merged**